conans/client/graph/profile_node_definer.py

```diff
@@ -26,8 +26,13 @@ def initialize_conanfile_profile(conanfile, profile_build, profile_host, base_co
     settings_build = _per_package_settings(conanfile, profile_build, ref)
     if is_build_require or base_context == CONTEXT_BUILD:
         _initialize_conanfile(conanfile, profile_build, settings_build.copy(), ref)
+        conanfile.buildenv_build = None
+        conanfile.conf_build = None
```

Comment:
Might be worth to define them in ``class ConanFile`` instead of here dynamically?

Reply:
What about `settings_build` and `settings_target`?

Reply:
Yes, you are right, I thought those had been defined in ``ConanFile`` already.
Looks good then, lets keep it this way.